core/src/main/java/org/elasticsearch/indices/query/IndicesQueriesRegistry.java

```diff
@@ -19,24 +19,47 @@
 
 package org.elasticsearch.indices.query;
 
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.ParseFieldMatcher;
+import org.elasticsearch.common.ParsingException;
 import org.elasticsearch.common.component.AbstractComponent;
 import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.XContentLocation;
 import org.elasticsearch.index.query.QueryParser;
 
 import java.util.Map;
 
 public class IndicesQueriesRegistry extends AbstractComponent {
     private Map<String, QueryParser<?>> queryParsers;
+    private Map<String, ParseField> parseFields;
+    private ParseFieldMatcher parseFieldMatcher;
 
-    public IndicesQueriesRegistry(Settings settings, Map<String, QueryParser<?>> queryParsers) {
+    public IndicesQueriesRegistry(Settings settings, Map<String, QueryParser<?>> queryParsers, Map<String, ParseField> parseFields) {
```

Comment:
It would be nice if we could somehow get the parseField from each parser, but I guess the problem is that QueryParser needs to be a functional interface? Maybe this can be added later.